### ISSUE TITLE
Keyboard is always registered as an input affecting split-screen games.

### ISSUE THREAD

**Issue by mikeymop** [ublue-os/bazzite#1345]
### Describe the bug

This issue is made apparent with games like Sackboys Great Adventure.

Traditionally this should be launched without the keyboard plugged in to correctly register a controller as player one.

Many issues can be remediated by removing the keyboard and remapping the controllers in steam input.
It seems in Bazzite, even with the keyboard unplugged, that a keyboard is registered as an input.

### What did you expect to happen?

Launching a game without a keyboard plugged in and a controller connected will default the controller to player one.

### Output of `rpm-ostree status`

Will provide shortly.

### Hardware

CPU: Ryzen 5 5500g
GPU: rx5500

**Comment by jsnyder** [comment]
I'm also experiencing this. It's also not clear to me in the game how to switch out the first player at all.

Does this have something to do with still having PS/2 ports on the motherboard for a keyboard?

**Comment by mikeymop** [comment]
I don't think so as the machine I am using does not have PS/2 ports.

What may be  happening is that the keyboard is mapped asto steam as the Steam Deck controls.

On the Steam Deck with Sackboy I also am forced to leave player 1 on the steam deck.

However on desktop Linux it works as expected for this game.

**Comment by jsnyder** [comment]
Are you saying that on a bog standard Steam Deck it sticks you with a keyboard player?  I don't remember that happening before (re-installing on my actual Deck now).

I'm also not sure what you mean by the keyboard being mapped to steam.  Is that essentially reflecting that the Steam Deck has it's built-in controls and therefore this version of Steam needs a replacement for that and that's set up as keyboard?

I have had issues with different games sometimes either getting stuck only taking keyboard input. It's kind of like some of the auto-input switching I've seen in other contexts like on Windows steam isn't working quite right.  Where if you hit a key/use the mouse you see all the in-game key/button layout help switch between modes.. now that doesn't happen and sometimes you get stuck with keyboard even though a controller is connected (this could be a separate issue).

I don't actually know a ton about how these input layers work on the various platforms and also Desktop vs Game Mode vs Big Picture for what to dig in on to resolve this.

**Comment by jsnyder** [comment]
Yeah, this does not happen on my stock standard Steam Deck (OLED):
![image](https://github.com/user-attachments/assets/57287c1c-7341-4057-bdeb-5bf20d4cf285)

(Nothing plugged in)

**Comment by mikeymop** [comment]
Hey,, thanks for the follow up.

I've tried again over the weekend on both games that affect me:

- Moonstone Island
- Sackboy: A Big Adventure

Both default to the keyboard as player one, even if no keyboard is plugged in when launching the games.


There is a workaround that works once per boot, where I start the game, and add a third controller while the game is running.

This exposes the "switch controller ordering" in Steam input ui. With which I can set the desired controller to `n+1` where `n` is the highest numbers virtual controller port in Steam Input.

If I relaunch the game, with the extra controller, them default pot will be `n+1` and I need a fourth controller to expose the port assignment and set player one to the next highest virtual port.

**Comment by mikeymop** [comment]
Thanks bot.

I'll test again and close if resolved otherwise I'll comment back.

**Comment by noelmiller** [comment]
Hi @mikeymop,

Unfortunately, I am not sure that there is a lot we can do on Bazzite to fix this since it is a title specific problem. I also have a similar problem with the new Binding of Isaac update where my controls are completely messed up (even on steam deck). 

My suggestion would be to reach out to the developers of the game and see if they are able to track down what might be causing this.
